### PR TITLE
Improve download interactions

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/vector-im/element-web/issues/18238
Fixes https://github.com/vector-im/element-web/issues/18239

With help from @Palid 

**See commits for individual descriptions.** Much of the detail is also contained within the diff itself.

The tooltip behaviour on the file name (see below) is technically an accident that it appears right over top of the tile, but I think it works out okay enough to not be concerned.

----

**Designers**: Instead of truncating to 20-30 characters, this truncates to roughly 15 characters so that the file size has a chance at fitting in the tile at all times.

----

image

image

image

----

Notes: none
element-web notes: none

### CODE REVIEW

File: src/utils/FileDownloader.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    public async download({ blob, name, autoDownload = true, opts = DEFAULT_STYLES }: DownloadOptions) {
+        const iframe = this.iframe; // get the iframe first just in case we need to await onload
+        if (this.onLoadPromise) await this.onLoadPromise;
+        iframe.contentWindow.postMessage({
+            ...opts,
+            blob: blob,
+            download: name,
+            auto: autoDownload,
+        }, '*');
```
Comment (gsouquet) [inline src/utils/FileDownloader.ts:101]: Should we use `location.origin` over `*`. It looks like that `iframe` will and should always be loaded on the same origin as the parent page

**Review by gsouquet — APPROVED** [review]
👏

File: src/utils/FileDownloader.ts (hunk 1 above)
Comment (turt2live) [reply src/utils/FileDownloader.ts:101]: possibly, though postMessage is known to be fiddly with origins so would advocate that we do it in a different PR

**Comment by janogarcia** [comment]
@turt2live 

Ideally the filename truncation should happen in the middle of the filename or a few characters before the filename extension, cause otherwise it's not going to be obvious for everyone that the three last chars are part of the extension name.

Example of filename truncation on macOS.
truncated-filenames

On the 2nd screenshot the expanded file name tooltip is obscuring the file size, which could be prevented by using a greater vertical offset.

On the 3rd screenshot the file size is also getting truncated, accidentally obscuring that data. 

In general, I think this component is in need of a much better design overall, so that the file extension and file size are not affected by the filename length, etc. But that's out of the scope of this PR.

**Comment by turt2live** [comment]
@janogarcia would it be okay to add the file size to the tooltip to cover all 3 issues for the short term?

**Comment by janogarcia** [comment]
@turt2live Let's do that for the short-term, yes.

**Comment by turt2live** [comment]
@janogarcia done; ptal :)

**Review by janogarcia — APPROVED** [review]

**Pull request merged**